src/ray/common/status.cc

```diff
@@ -31,88 +31,57 @@
-#define STATUS_CODE_TRANSIENT_OBJECT_STORE_FULL "TransientObjectStoreFull"
-#define STATUS_CODE_OUT_OF_DISK "OutOfDisk"
-#define STATUS_CODE_OBJECT_UNKNOWN_OWNER "ObjectUnknownOwner"
-#define STATUS_CODE_RPC_ERROR "RpcError"
-#define STATUS_CODE_OUT_OF_RESOURCE "OutOfResource"
-#define STATUS_CODE_OBJECT_REF_END_OF_STREAM "ObjectRefEndOfStream"
-#define STATUS_CODE_AUTH_ERROR "AuthError"
-#define STATUS_CODE_INVALID_ARGUMENT "InvalidArgument"
-#define STATUS_CODE_CHANNEL_ERROR "ChannelError"
-#define STATUS_CODE_CHANNEL_TIMEOUT_ERROR "ChannelTimeoutError"
-
+constexpr std::string_view kStatusCodeOk = "OK";
 // not a real status (catch all for codes not known)
-#define STATUS_CODE_UNKNOWN "Unknown"
+constexpr std::string_view kStatusCodeUnknown = "Unknown";
 
 namespace {
 
 // Code <-> String mappings.
 
-const absl::flat_hash_map<StatusCode, std::string> kCodeToStr = {
-    {StatusCode::OK, STATUS_CODE_OK},
-    {StatusCode::OutOfMemory, STATUS_CODE_OUT_OF_MEMORY},
-    {StatusCode::KeyError, STATUS_CODE_KEY_ERROR},
-    {StatusCode::TypeError, STATUS_CODE_TYPE_ERROR},
-    {StatusCode::Invalid, STATUS_CODE_INVALID},
-    {StatusCode::IOError, STATUS_CODE_IO_ERROR},
-    {StatusCode::UnknownError, STATUS_CODE_UNKNOWN_ERROR},
-    {StatusCode::NotImplemented, STATUS_CODE_NOT_IMPLEMENTED},
-    {StatusCode::RedisError, STATUS_CODE_REDIS_ERROR},
-    {StatusCode::TimedOut, STATUS_CODE_TIMED_OUT},
-    {StatusCode::Interrupted, STATUS_CODE_INTERRUPTED},
-    {StatusCode::IntentionalSystemExit, STATUS_CODE_INTENTIONAL_SYSTEM_EXIT},
-    {StatusCode::UnexpectedSystemExit, STATUS_CODE_UNEXPECTED_SYSTEM_EXIT},
-    {StatusCode::CreationTaskError, STATUS_CODE_CREATION_TASK_ERROR},
-    {StatusCode::NotFound, STATUS_CODE_NOT_FOUND},
-    {StatusCode::Disconnected, STATUS_CODE_DISCONNECTED},
-    {StatusCode::SchedulingCancelled, STATUS_CODE_SCHEDULING_CANCELLED},
-    {StatusCode::AlreadyExists, STATUS_CODE_ALREADY_EXISTS},
-    {StatusCode::ObjectExists, STATUS_CODE_OBJECT_EXISTS},
-    {StatusCode::ObjectNotFound, STATUS_CODE_OBJECT_NOT_FOUND},
-    {StatusCode::ObjectAlreadySealed, STATUS_CODE_OBJECT_ALREADY_SEALED},
-    {StatusCode::ObjectStoreFull, STATUS_CODE_OBJECT_STORE_FULL},
-    {StatusCode::TransientObjectStoreFull, STATUS_CODE_TRANSIENT_OBJECT_STORE_FULL},
-    {StatusCode::OutOfDisk, STATUS_CODE_OUT_OF_DISK},
-    {StatusCode::ObjectUnknownOwner, STATUS_CODE_OBJECT_UNKNOWN_OWNER},
-    {StatusCode::RpcError, STATUS_CODE_RPC_ERROR},
-    {StatusCode::OutOfResource, STATUS_CODE_OUT_OF_RESOURCE},
-    {StatusCode::ObjectRefEndOfStream, STATUS_CODE_OBJECT_REF_END_OF_STREAM},
-    {StatusCode::AuthError, STATUS_CODE_AUTH_ERROR},
-    {StatusCode::InvalidArgument, STATUS_CODE_INVALID_ARGUMENT},
-    {StatusCode::ChannelError, STATUS_CODE_CHANNEL_ERROR},
-    {StatusCode::ChannelTimeoutError, STATUS_CODE_CHANNEL_TIMEOUT_ERROR},
+const absl::flat_hash_map<StatusCode, std::string_view> kCodeToStr = {
```

Comment:
(no action needed) for these maps known at compile-time, one common way is `StaticMap`, which is internally a `std::array<>`.

Reply:
A map that is determined at compile time, usually implemented in array. its not part of stl standard nor abseil, which we need to implement ourselves

Reply:
Got it. Thanks!